### PR TITLE
compiler: introduce and use `MType::is_c_primitive`

### PR DESCRIPTION
**Pull request by privat**
Thus remove all comparaison to "val*" in the code, this is cleaner.

### CODE REVIEW

**Comment by xymus** [comment]
:shipit:

**Comment by jpages** [comment]
+1

File: src/compiler/abstract_compiler.nit [hunk 1]
```diff
@@ -1754,12 +1754,16 @@ redef class MType
 
 	# Short name of the `ctype` to use in unions
 	fun ctypename: String do return "val"
+
+	# Is the associated C type a primitive one?
+	#
+	# ASSERT `result == (ctype == "val*")`
```
Comment (Morriar) [inline src/compiler/abstract_compiler.nit]: The ASSERT is a lie.

File: src/compiler/abstract_compiler.nit [hunk 2]
```diff
@@ -1777,6 +1781,25 @@ redef class MClassType
...
+			return true
+		else if mclass.name == "Bool" then
+			return true
+		else if mclass.name == "Char" then
+			return true
+		else if mclass.name == "Float" then
+			return true
+		else if mclass.name == "NativeString" then
+			return true
+		else if mclass.name == "NativeArray" then
```
Comment (Morriar) [inline src/compiler/abstract_compiler.nit]: Is this for understanding purpose? This branch shjould be merged in the `else`.

Comment (Morriar) [inline src/compiler/abstract_compiler.nit]: Actually, when I read this quickly my eyes automatically assumed that "NativeArray" returned `true`.

**Comment by Morriar** [comment]
+1

**Comment by privat** [comment]
reroll: fixed the lie and the implem (pourquoi faire simple quand on peut faire compliqué)

**Pull request merged**